pkg/proto/nbcontract/v1/vmtype.proto

```diff
@@ -0,0 +1,7 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+enum VmType {
+  VM_TYPE_STANDARD = 0;
```

Comment:
Not sure if defaulting to standard is the best move here, but this is what AB currently does unless vmss is explicity specified.

Reply:
Agreed. I think it should be good as long as AB is doing this too.

Reply:
BTW, protobuff official doc suggests to set the first enum=0 as unspecified to handle evolution scenario.
You may want to consider unspecified = 0, standard = 1, etc. And in our go binary parser, if it's unspecified, we will still set it to standard. Does that sound make sense to you?
https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum

Reply:
Do we know where (and what for) this is used?

Reply:
the only place I found is that is it is logged into the azure json file: https://github.com/Azure/AgentBaker/blob/master/parts/linux/cloud-init/artifacts/cse_config.sh#L192C5-L192C28

i'm not sure how this file is used however

Reply:
This file is used by cloud-provder-azure clients, the structure is documented here: https://cloud-provider-azure.sigs.k8s.io/install/configs/. But:
* Cataloguing all of the uses of this _on the node_ would be great. Currently it is likely used by kubelet (to authenticate to ACRs, via in-tree auth helpers which use cloud-provider-azure clients) and probably CSI drivers. They may be other uses I am not aware of.
* I am not sure why any of these need vmType ... (kubelet ACR auth definitely should not need it?)
* From the contract perspective, it may still be a good idea to collect everything that goes to azure.json in one place (unless we have a clear indication some settings are used elsewhere as well)

Reply:
Sure I can start noting down where azure.json is used and what variables in the contract are stored inside.

When you say collect them in one place, do you mean grouping all the variables in azure.json into one config in the contract?